distribution/tools/server-cli/src/test/java/org/elasticsearch/server/cli/APMJvmOptionsTests.java

```diff
@@ -62,6 +68,53 @@ public void testFileDeleteWorks() throws IOException {
         assertFalse(Files.exists(tempFile));
     }
 
+    public void testExtractSettings() throws UserException {
+        Settings settings = Settings.builder()
+            .put("tracing.apm.enabled", true)
+            .put("tracing.apm.agent.server_url", "https://myurl:443")
+            .put("tracing.apm.agent.service_node_name", "instance-0000000001")
+            .put("tracing.apm.agent.global_labels.deployment_id", "123")
+            .put("tracing.apm.agent.global_labels.deployment_name", "APM Tracing")
+            .put("tracing.apm.agent.global_labels.organization_id", "456")
+            .build();
+
+        var extracted = APMJvmOptions.extractApmSettings(settings);
+
+        assertThat(
+            extracted,
+            allOf(
+                hasEntry("server_url", "https://myurl:443"),
+                hasEntry("service_node_name", "instance-0000000001"),
+                hasKey("global_labels"), // test that we have collapsed all global labels into one
+                not(hasKey("global_labels.organization_id")) // tests that we strip out the top level label keys
+            )
+        );
+        assertThat(
+            extracted.get("global_labels"),
+            allOf(containsString("deployment_name=APM Tracing"), containsString("organization_id=456"), containsString("deployment_id=123"))
```

Comment:
We're not actually testing that the string we pass to the APM agent is comma-separated. Do you think we should check that?

Reply:
Yes very good point, I couldn't test the string directly because the ordering is unpredictable, but I'll try to add another way.

Reply:
You could at least try splitting on commas and asserting that you have the number of elements that you expect? Or split then sort? I would have hoped that Hamcrest matchers would let you match in any order too, but perhaps not.